pkg/util/iptables/save_restore.go

```diff
@@ -21,102 +21,32 @@ import (
 	"fmt"
 )
 
-var (
-	commitBytes = []byte("COMMIT")
-	spaceBytes  = []byte(" ")
-)
-
 // MakeChainLine return an iptables-save/restore formatted chain line given a Chain
 func MakeChainLine(chain Chain) string {
 	return fmt.Sprintf(":%s - [0:0]", chain)
 }
 
-// GetChainLines parses a table's iptables-save data to find chains in the table.
-// It returns a map of iptables.Chain to []byte where the []byte is the chain line
-// from save (with counters etc.).
-// Note that to avoid allocations memory is SHARED with save.
-func GetChainLines(table Table, save []byte) map[Chain][]byte {
-	chainsMap := make(map[Chain][]byte)
-	tablePrefix := []byte("*" + string(table))
-	readIndex := 0
-	// find beginning of table
-	for readIndex < len(save) {
-		line, n := readLine(readIndex, save)
-		readIndex = n
-		if bytes.HasPrefix(line, tablePrefix) {
+// GetChains parses iptables-save data to find the chains that are defined.
+// It assumes that save contains a single table's data, and returns a map with
+// keys for every chain defined in that table.
+func GetChains(save []byte) map[Chain]struct{} {
+	chainsMap := make(map[Chain]struct{})
+
+	for {
+		i := bytes.Index(save, []byte("\n:"))
```

Comment:
the logic is much simpler now, just get everything between `\n:`  and a blank space